### PR TITLE
use go plugin for sparkoneks, hivemetastore and rdsdatabase

### PR DESCRIPTION
**Pull request by anson627**
@hiboyang

### CODE REVIEW

**Comment by hiboyang** [comment]
+1 great work!

File: pkg/topologyimpl/hivemetastore/topology.go [hunk 1]
```diff
@@ -14,10 +14,11 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package hivemetastore
+package main
```
Comment (hiboyang) [inline pkg/topologyimpl/hivemetastore/topology.go:17]: after this pr, may refactor the package and put these plugin code in pkg/plugin/hivemetastore?

**Review by hiboyang — APPROVED** [review]

File: pkg/topologyimpl/hivemetastore/topology.go (hunk 1 above)
Comment (anson627) [reply pkg/topologyimpl/hivemetastore/topology.go:17]: yes, refactoring my next step, plugin code should be separated from topology implementation

File: cmd/punch/root.go [hunk 2]
```diff
@@ -69,10 +71,37 @@ func Execute() {
...
 }
 
-func getTopologyHandlerOrFatal(kind string) framework.TopologyHandler {
-	handler := framework.DefaultTopologyHandlerManager.GetHandler(kind)
-	if handler == nil {
-		log.Fatalf("Topology kind %s not supported", kind)
+func getTopologyHandlerPlugin(kind string) framework.TopologyHandler {
+	var mod string
+	switch kind {
+	case "SparkOnEks":
```
[the remark this answers is not in the document]
Comment (anson627) [reply cmd/punch/root.go:77]: that's one way, the other way is to use kind to infer plugin file name and then try to load the plugin with file name in the folder

**Pull request merged**